packages/account-export/src/lib/helpers/importAccount.ts

```diff
@@ -0,0 +1,86 @@
+import generator from "generate-password";
+import nacl from "tweetnacl";
+import {
+  decodeUTF8,
+  encodeUTF8,
+  encodeBase64,
+  decodeBase64,
+} from "tweetnacl-util";
+
+import type { AccountImportData } from "@near-wallet-selector/core";
+
+interface DecryptAccountDataProps {
+  ciphertext: string;
+  secretKey: string;
+}
+
+interface EncryptAccountDataProps {
+  accountData: Array<AccountImportData>;
+  secretKey: string;
+}
+
+export const encryptAccountData = ({
+  accountData,
+  secretKey,
+}: EncryptAccountDataProps): string => {
+  if (!secretKey) {
+    throw new Error("Secret key is required");
+  }
+  try {
+    const keyUint8Array = decodeBase64(
+      Buffer.from(secretKey).toString("base64")
+    );
+    const messageUint8Array = decodeUTF8(JSON.stringify(accountData));
+    const nonce = nacl.randomBytes(nacl.secretbox.nonceLength);
+    const box = nacl.secretbox(messageUint8Array, nonce, keyUint8Array);
+    const fullMessage = new Uint8Array(nonce.length + box.length);
+    fullMessage.set(nonce);
+    fullMessage.set(box, nonce.length);
+
+    const base64FullMessage = encodeBase64(fullMessage);
+    return base64FullMessage;
+  } catch (e) {
+    throw new Error("Unable to encrypt account data");
+  }
+};
+
+export const decryptAccountData = ({
+  ciphertext,
+  secretKey,
+}: DecryptAccountDataProps): Array<AccountImportData> => {
+  if (!secretKey) {
+    throw new Error("Secret key is required");
+  }
+  try {
+    const keyUint8Array = decodeBase64(
+      Buffer.from(secretKey).toString("base64")
+    );
+    const messageWithNonceAsUint8Array = decodeBase64(ciphertext);
+    const nonce = messageWithNonceAsUint8Array.slice(
+      0,
+      nacl.secretbox.nonceLength
+    );
+    const message = messageWithNonceAsUint8Array.slice(
+      nacl.secretbox.nonceLength,
+      ciphertext.length
+    );
+    const decrypted = nacl.secretbox.open(message, nonce, keyUint8Array);
+    if (!decrypted) {
+      throw new Error("Unable to decrypt account data");
+    }
+    const base64DecryptedMessage = encodeUTF8(decrypted);
+    return JSON.parse(base64DecryptedMessage);
+  } catch {
+    throw new Error("Unable to decrypt account data");
+  }
+};
+
+export const generateSecretKey = (): string =>
+  generator.generate({
+    length: 32,
```

Comment:
Only 32 char-length strings work as values here, pretty sure this is why... Let's set this using the library's value https://github.com/dchest/tweetnacl-js/blob/master/README.md#naclsecretboxkeylength--32

```suggestion
    length: nacl.secretbox.keyLength,
```